Sources/TuistKit/Services/RunService.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import RxSwift
+import TSCBasic
+import TuistAutomation
+import TuistCore
+import TuistSupport
+
+enum RunServiceError: FatalError {
+    case appNotFound(targetName: String)
+    case simulatorNotFound
+    case targetNotFound(forSchemeName: String)
+
+    var description: String {
+        switch self {
+        case let .appNotFound(targetName):
+            return "Couldn't find any apps built for \(targetName)"
+        case .simulatorNotFound:
+            return "Could't find simulator"
+        case let .targetNotFound(forSchemeName):
+            return "Couldn't find \(forSchemeName)"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .appNotFound, .simulatorNotFound, .targetNotFound:
+            return .abort
+        }
+    }
+}
+
+final class RunService {
+    let simulatorController: SimulatorControlling
+
+    let buildGraphInspector: BuildGraphInspecting
+
+    let projectGenerator: ProjectGenerating
+
+    let xcodebuildController: XcodeBuildControlling
+
+    var appPath: AbsolutePath?
+
+    var buildableTarget: Target?
+
+    init(simulatorController: SimulatorControlling = SimulatorController(),
+         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
+         projectGenerator: ProjectGenerating = ProjectGenerator(),
+         xcodebuildController: XcodeBuildControlling = XcodeBuildController())
+    {
+        self.simulatorController = simulatorController
+        self.buildGraphInspector = buildGraphInspector
+        self.projectGenerator = projectGenerator
+        self.xcodebuildController = xcodebuildController
+    }
+
+    func run(schemeName: String) throws {
+        try buildScheme(schemeName)
+
+        guard let buildableTarget = buildableTarget else {
+            throw RunServiceError.targetNotFound(forSchemeName: schemeName)
+        }
+
+        appPath = try findApp(for: buildableTarget)
+        try launchApp(for: buildableTarget)
```

Comment:
Since you have the build process, which you can model with an `Observable`, and the simulator logic, which can also be wrapped inside an `Observable`, I think there's an opportunity here to do things in parallel.

1. You trigger the build of the app and the launch of the simulator in parallel.
2. If the build finishes successfully, then you install and launch the app.

I think you can model the logic with 3 observables:

1. buildAppObservable
2. setupDeviceObservable
3. installAndLaunchAppObservable.

1 and 2 can run in parallel, while 3 has to wait until the first 2 finish successfully.